ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+            condition=constants.STATUS_RUNNING,
+            selector=constants.NOOBAA_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.RGW_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.PROVIDER_SERVER_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.RGW_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        list_of_rgw_pods = pod.get_rgw_pods(
+            namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        rgw_pod_obj = list_of_rgw_pods[0]
+        restart_count_for_rgw_pod = pod.get_pod_restarts_count(
+            list_of_pods=list_of_rgw_pods,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        rgw_pod_restart_count = restart_count_for_rgw_pod[rgw_pod_obj.name]
+        log.info(f"restart count for rgw pod is: {rgw_pod_restart_count}")
+        assert (
+            restart_count_for_rgw_pod[rgw_pod_obj.name] == 0
+        ), f"Error rgw pod has restarted {rgw_pod_restart_count} times"
+
+        # Check ocs-storagecluster is in 'Ready' status
+        log.info("Verify storagecluster on Ready state")
+        verify_storage_cluster()
+
+        # Check backing storage is s3-compatible
+        backingstore_type = check_pv_backingstore_type()
+        log.info(f"backingstore value: {backingstore_type}")
+        assert backingstore_type == constants.BACKINGSTORE_TYPE_S3_COMP
+
+    def odf_subscription_on_provider(self):
+        """
+        This method creates odf subscription for the provider
+        """
+        # Check if odf is available already on the provider
+        ceph_cluster = ocp.OCP(
+            kind="CephCluster", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        try:
+            ceph_cluster.get().get("items")[0]
+            log.info("OCS cluster already exists")
+            return
+        except (IndexError, CommandFailed):
+            log.info("Running ODF subscription for the provider")
+
+        live_deployment = config.DEPLOYMENT.get("live_deployment")
+        if not live_deployment:
+            create_catalog_source()
+
+        log.info("Creating namespace and operator group.")
+        olm_data = templating.load_yaml(constants.OLM_YAML)
+        templating.dump_data_to_temp_yaml(olm_data, constants.OLM_YAML)
+        run_cmd(f"oc create -f {constants.OLM_YAML}")
+        self.deployment.subscribe_ocs()
+
+        # Wait until odf is installed
+        odf_operator = defaults.ODF_OPERATOR_NAME
+        self.deployment.wait_for_subscription(
```

Comment:
`self.wait_for_csv(ocs_operator_name)` and `self.wait_for_subscription(ocs_operator_name)` are already a part of  `self.deployment.subscribe_ocs()`. We'll need to remove it to avoid making this twice.

Reply:
Addressed in latest commit.